src/App.jsx

```diff
@@ -185,163 +186,165 @@ function App() {
   };
 
   return (
-    <div className="app">
-      <Navbar
-        row={rowMode}
-        setRow={setRowMode}
-        user={user}
-        setUser={setUser}
-        open={open}
-        setOpen={setOpen}
-        setLogout={setLogout}
-      />
-
-      <ShareModal
-        openShareModal={openShareModal}
-        setOpenShareModal={setOpenShareModal}
-        currentPostLink={currentPostLink}
-        postText={postText}
-      />
-
-      <Modal open={logout} onClose={() => setLogout(false)}>
-        <div style={getModalStyle()} className={classes.paper}>
-          <form className="modal__signup">
-            <img
-              src="https://user-images.githubusercontent.com/27727921/185767526-a002a17d-c12e-4a6a-82a4-dd1a13a5ecda.png"
-              alt="dummygram"
-              className="modal__signup__img"
-              style={{
-                width: "80%",
-                marginLeft: "10%",
-                filter: "invert(var(--val))",
-              }}
-            />
+    <RowModeContext.Provider value={rowMode}>
```

Comment:
instead of defining the context at app, can you define it at index/main where other contexts are defined as well?

Reply:
that way we won't have to pass it as a prop to navbar as well

Reply:
So, you are saying that I should wrap the app with context provider in main file like in the screenshot and update it in navbar. Right?

Reply:
yess